generators/typescript/sdk/request-wrapper-generator/src/GeneratedRequestWrapperImpl.ts

```diff
@@ -198,6 +225,138 @@ export class GeneratedRequestWrapperImpl implements GeneratedRequestWrapper {
         };
     }
 
+    private getTypeForBodyProperty(
+        requestBody: InlinedRequestBody | FileUploadRequest,
+        property: InlinedRequestBodyProperty,
+        context: SdkContext
+    ): TypeReferenceNode {
+        const propParentTypeName = requestBody.name.pascalCase.safeName;
+        const propName = property.name.name.pascalCase.safeName;
+        return context.type.getReferenceToInlinePropertyType(property.valueType, propParentTypeName, propName);
+    }
+
+    private generateModule(
+        inlinedRequestBody: InlinedRequestBody,
+        context: SdkContext
+    ): ModuleDeclarationStructure | undefined {
+        if (!this.inlineInlineTypes) {
+            return undefined;
+        }
+
+        const inlineProperties = this.getInlinePropertiesWithTypeDeclaration(inlinedRequestBody, context);
+        if (inlineProperties.size === 0) {
+            return;
+        }
+        return {
+            kind: StructureKind.Module,
+            name: inlinedRequestBody.name.pascalCase.safeName,
+            isExported: true,
+            hasDeclareKeyword: false,
+            declarationKind: ModuleDeclarationKind.Namespace,
+            statements: Array.from(inlineProperties.entries()).flatMap(
+                ([objectProperty, typeDeclaration]: [ObjectProperty, TypeDeclaration]) => {
+                    const typeName = objectProperty.name.name.pascalCase.safeName;
+                    const listOrSetStatementGenerator = () => {
```

Comment:
Similar to above w.r.t. closures. But I think all of this will be resolved based on your suggestion to split this out into a shared component.